python_modules/dagster/dagster/_core/definitions/asset_graph.py

```diff
@@ -185,11 +194,18 @@ def from_assets(
                 freshness_policies_by_key.update(asset.freshness_policies_by_key)
                 auto_materialize_policies_by_key.update(asset.auto_materialize_policies_by_key)
                 backfill_policies_by_key.update({key: asset.backfill_policy for key in asset.keys})
-                if len(asset.keys) > 1 and not asset.can_subset:
-                    for key in asset.keys:
-                        required_multi_asset_sets_by_key[key] = asset.keys
                 code_versions_by_key.update(asset.code_versions_by_key)
 
+                if not asset.can_subset:
+                    if len(asset.keys) > 1:
```

Comment:
the next diff removes required_multi_asset_sets_by_key so this block gets halved https://github.com/dagster-io/dagster/pull/16928/files#diff-5dd8d9eba33f4d3fdd9c092dafeff8a34553362f9765246b0baaf9942be63487L200